go.mod

```diff
@@ -1,6 +1,8 @@
 module github.com/mongodb/mongodb-atlas-cli
 
-go 1.20
+go 1.21
+
+toolchain go1.21.3
```

Comment:
Are there any reasons why users cannot use the toolchain 1.21.0 / 1.21.1 / 1.21.2?  if not, I would remove this

Reply:
According to https://go.dev/doc/toolchain if you run using 1.21.0-2 or older like 1.20 the toolchain will automatically download go 1.21.3 compile it and continue it's flow, if running 1.21.4+ or even 1.22 it will continue believing the compatibility promise.

I think this is a good change.

Reply:
This is a good pattern to align toolchain versions across asdf, evergreen, and github actions. 
Go toolchains add an additional layer of go versioning so I wanted to be more explicit to ensure alignment.

Context: 
https://go.dev/blog/toolchain#:~:text=If%20you%20are,the%20module%20cache.

I have no strong opinion to keep it. This way we can also align other dependencies like Kubernetes operator etc.

Why:

With 1.21 we cannot anymore use 1.20 as a universal binary that always points to latest and tools like ads need to specifically specify the patch version:

Non Patch
```
asdf install golang 1.21
URL: https://dl.google.com/go/go1.21.darwin-arm64.tar.gz returned status 404
```
With patch:
```
▶ asdf install golang 1.21.3            
golang 1.21.3 is already installed
```

Reply:
> I think this is a good change.
Does this force us to update the toolversion for each new patch version of 1.21? If there is no reason we cannot run the CLI with previous patch versions of 1.21, I don't understand why we need this 🤔

Reply:
@andreaangiolillo this is not blocking old versions (like 1.21.1) it is actually improving the scenario (if you run on a very old version you will still compile on a newer compiler).

Reply:
> @andreaangiolillo this is not blocking old versions (like 1.21.1) it is actually improving the scenario (if you run on a very old go version you will still compile on a newer compiler).

I understood was not blocking older versions. I am wondering if we will need to keep updating the toolchain version for each new patch release of 1.21. 

Btw my comment is not blocking, I am just trying to understand how we want to maintain the toolchain version going forward and if there was a specific reason to select 1.21.3 besides being the latest patch release.

Reply:
since toolchain was removed we can leave it for now

Reply:
My assumption was to update it together with go versions

Reply:
>This is a good pattern to align toolchain versions across asdf, evergreen, and github actions.

Not necessarily true, for example right now packaging happens in evg (we won't change that due to notarization/signing), up until this change we were using the latest patch version when building distributables, which we won't do anymore, also evergreen has usually discouraged from pining to a patch for this same reason

IMHO the new toolchain feature replaces asdf and we should consider deprecating from this repo, we also need to decide what to do about evg, do we need to keep a reminder/task to keep updating patch versions to make sure we package with included security fixes or we should stick to the current evg recommended pattern to not pin to a minor version